langchain/llms/beam.py

```diff
@@ -0,0 +1,299 @@
+                "transformers",
+                "torch",
+                "pillow",
+                "accelerate",
+                "safetensors",
+                "xformers",],
+            max_length=50)
+
+        depolyResult = llm._deploy(prompt=input)
+
+        callResult = llm._call(prompt=input)
+    """
+
+    model_name: str = ""
+    name: str = ""
+    cpu: str = ""
+    memory: str = ""
+    gpu: str = ""
+    python_version: str = ""
+    python_packages: List[str] = []
+    max_length: str = ""
+    url: str = ""
+    """model endpoint to use"""
+
+    model_kwargs: Dict[str, Any] = Field(default_factory=dict)
+    """Holds any model parameters valid for `create` call not
+    explicitly specified."""
+
+    beam_client_id: str = ""
+    beam_client_secret: str = ""
+
+    class Config:
+        """Configuration for this pydantic config."""
+
+        extra = Extra.forbid
+
+    @root_validator(pre=True)
+    def build_extra(cls, values: Dict[str, Any]) -> Dict[str, Any]:
+        """Build extra kwargs from additional params that were passed in."""
+        all_required_field_names = {field.alias for field in cls.__fields__.values()}
+
+        extra = values.get("model_kwargs", {})
+        for field_name in list(values):
+            if field_name not in all_required_field_names:
+                if field_name in extra:
+                    raise ValueError(f"Found {field_name} supplied twice.")
+                logger.warning(
+                    f"""{field_name} was transfered to model_kwargs.
+                    Please confirm that {field_name} is what you intended."""
+                )
+                extra[field_name] = values.pop(field_name)
+        values["model_kwargs"] = extra
+        return values
+
+    @root_validator()
+    def validate_environment(cls, values: Dict) -> Dict:
+        """Validate that api key and python package exists in environment."""
+        beam_client_id = get_from_dict_or_env(
+            values, "beam_client_id", "BEAM_CLIENT_ID"
+        )
+        beam_client_secret = get_from_dict_or_env(
+            values, "beam_client_secret", "BEAM_CLIENT_SECRET"
+        )
+        values["beam_client_id"] = beam_client_id
+        values["beam_client_secret"] = beam_client_secret
+        return values
+
+    @property
+    def _identifying_params(self) -> Mapping[str, Any]:
+        """Get the identifying parameters."""
+        return {
+            **{"model_name": self.model_name},
+            **{"name": self.name},
+            **{"cpu": self.cpu},
+            **{"memory": self.memory},
+            **{"gpu": self.gpu},
+            **{"python_version": self.python_version},
+            **{"python_packages": self.python_packages},
+            **{"max_length": self.max_length},
+            **{"model_kwargs": self.model_kwargs},
+        }
+
+    @property
+    def _llm_type(self) -> str:
+        """Return type of llm."""
+        return self.model_name
```

Comment:
this should correspond to the class type not the specific model (e.g. "beam" in this case)